### DISCUSSION TITLE
Add "filter" option to m2o/o2m/m2m

### DISCUSSION THREAD

**Discussion by rijkvanzanten** [directus/directus#5185] [Feature Requests]
Allows you to scope what users can select from

**Comment by Michael-Nussbaumer** [comment]
It would be great to consider a combination with https://github.com/directus/directus/discussions/2993 to make it possbile to have field related filter.
 
For Example show just states of the selected country. :)

**Comment by rijkvanzanten** [comment]
Yes! That's a very good idea

**Comment by magomimmo** [comment]
> Yes! That's a very good idea

Agree. It's a very common scenario. In the mean time have you any workaround?

**Comment by benhaynes** [comment]
@magomimmo — yes, a custom interface 😉

**Comment by magomimmo** [comment]
Thanks @benhaynes, I appreciate

**Comment by bernatvadell** [comment]
Hi @rijkvanzanten 

I have posted a PR #6834 with the implementation of this conversation.  It is still in draft, but I would like you to take a look to see if it is the right way to go.

at the moment I have m2o implemented.

Ty

**Comment by wc-matteo** [comment]
and sorting as well: https://github.com/directus/directus/discussions/9087

**Comment by JamesW1** [comment]
Closing this request as this is now available in the product 🎉
